js/src/forum/states/PostStreamState.js

```diff
@@ -280,7 +280,13 @@ class PostStreamState {
         }
       });
 
-    return loadIds.length ? app.store.find('posts', loadIds) : Promise.resolve(loaded);
+    if (loadIds.length) {
+      return app.store.find('posts', loadIds).then((newPosts) => {
+        return loaded.concat(newPosts).sort((a, b) => a.id() - b.id());
```

Comment:
Posts should be sorted by number, not ID. An ext like e.g. fof/merge-discussions would break this.

Did the posts previously not get sorted in the JS (before rewrite)? 🤔

Reply:
Posts are sorted by id in DiscussionPage (although that is probably redundant). I think as long as fof/merge-discussions sorts posts by creation_date and does not alter the creation date, it should be fine. Otherwise the backend APIs would also need to be adjusted to calculate the (`near`) offset based on the post number rather than the date.

Reply:
That doesn't make sense then 🤔 we have a number column, why is it not being used for the sorting?

Also merge-discussions sets the number column based on creation I think, so I guess that's basically the same thing as ID.

Maybe this is not an issue. I think I had just woken up when I commented 🤔 not entirely sure.

Reply:
I would prefer to use `created_at` (or `number`) for sorting, as those columns are there to explicitly represent post order. `id` is ordered, but that's not one of it's purposes

Reply:
Agree, let's keep consistent with how the backend sorts posts. I updated it to sort by creation date.

Reply:
Hm still don't like that it's creation date, but that's a separate issue.

Reply:
Any specific concerns about creation date? Although now that I think about it, doesn't this mean that if 2 posts are created at the same time, their ordering might vary from time to time? From that perspective, `number` might be better, at least to keep things constant.

Reply:
I remember, that there is an admin API option to modify the creation date when creating posts. So post numbers can be in a different order than the ids (and creation dates).

Reply:
I'm just saying that sorting in the frontend instead of by the defined numbering system we already have set up is a separate issue that should be addressed separate from this PR.

Reply:
I know. I'm saying that this is a separate issue that we should address at another point.

I'm trying to stop the conversation about it in this PR but you just keep continuing it 🙂

Reply:
Alright, sorry for the confusion. If you're good with the PR, could you approve it?